### PR TITLE
fix version issue in datasource mso_site

### PR DESCRIPTION
**Pull request by shrsr**
Add option to accommodate v1 and v2 for mso_site datasource

### CODE REVIEW

**Review by lhercot — DISMISSED** [review]
LGTM

File: mso/datasource_mso_site.go [hunk 1]
```diff
@@ -86,67 +87,117 @@ func datasourceMSOSiteRead(d *schema.ResourceData, m interface{}) error {
...
+			if val.(map[string]interface{})["name"].(string) == name {
+				flag = true
+				break
+			}
+		} else {
+			val := info.(map[string]interface{})
+			if val["name"].(string) == name {
+				flag = true
+				break
+			}
```
Comment (lhercot) [inline mso/datasource_mso_site.go:119]: I would try to rewrite this as follow.

```
val := info.(map[string]interface{})
if platform == "nd" {
    val = val["common"].(map[string]interface{})
} 
if val["name"].(string) == name {
    flag = true
    break
}
```

File: mso/datasource_mso_site.go [hunk 2]
```diff
@@ -86,67 +87,117 @@ func datasourceMSOSiteRead(d *schema.ResourceData, m interface{}) error {
...
 	}
 
 	if flag != true {
 		return fmt.Errorf("Site of specified name not found")
 	}
 
-	dataCon := con.S("sites").Index(count)
-
+	var dataCon *container.Container
+	dataCon = con.S("sites").Index(count)
```
Comment (lhercot) [inline mso/datasource_mso_site.go:130]: can't you do dataCon := con.S("sites").Index(count) to replace those two lines?

File: mso/datasource_mso_site.go [hunk 3]
```diff
@@ -86,67 +87,117 @@ func datasourceMSOSiteRead(d *schema.ResourceData, m interface{}) error {
...
 	d.SetId(models.StripQuotes(dataCon.S("id").String()))
 
-	d.Set("name", models.StripQuotes(dataCon.S("name").String()))
+	if platform == "nd" {
+		dataCon_attr := dataCon.S("common")
```
Comment (lhercot) [inline mso/datasource_mso_site.go:134]: dataCon_attr -> dataConAttr

**Review by lhercot — CHANGES_REQUESTED** [review]

File: mso/datasource_mso_site.go (hunk 3 above)
Comment (shrsr) [reply mso/datasource_mso_site.go]: Done!

File: mso/datasource_mso_site.go (hunk 2 above)
Comment (shrsr) [reply mso/datasource_mso_site.go]: Yes!

File: mso/datasource_mso_site.go (hunk 1 above)
Comment (shrsr) [reply mso/datasource_mso_site.go]: Cool! Done!

**Review by lhercot — APPROVED** [review]
LGTM

**Pull request merged**